tests/validators/test_function.py

```diff
@@ -916,35 +1011,6 @@ def f_w(v: Any, handler: core_schema.ValidatorFunctionWrapHandler, info: core_sc
     calls.clear()
 
 
-def test_reprs() -> None:
-    reprs: list[str] = []
-
-    def sample_repr(v: Any, info: core_schema.ValidationInfo) -> Any:
-        reprs.append(repr(info))
-        return v
-
-    v = SchemaValidator(
-        core_schema.chain_schema(
-            [
-                core_schema.with_info_plain_validator_function(sample_repr),
-                core_schema.with_info_plain_validator_function(sample_repr, field_name='x'),
-            ]
-        )
-    )
-
-    class Foo:
-        def __repr__(self) -> str:
-            return 'This is Foo!'
-
-    v.validate_python(Foo())
-
-    # insert_assert(reprs)
-    assert reprs == [
-        'ValidationInfo(config=None, context=None, data=None, field_name=None)',
-        "ValidationInfo(config=None, context=None, data=None, field_name='x')",
```

Comment:
This test explicitly tested the now-deprecated behavior of setting a `field_name` on the validator function schema. If we assume no one was directly constructing these instead of us, removing this should be fine.

I could also leave the test and wrap the call in `pytest.warns`, to verify the old behavior keeps working (which it does) -- let me know what you prefer.